src/CrossValidation.cpp

```diff
@@ -242,226 +283,192 @@ int CrossValidation::doStep(bool updateDOC, Normalizer* pNorm, double selectionF
       testScores_[set].setDOCFeatures(pNorm);
     }
   }
+
+  // Below implements the series of speedups detailed in the following:
+  // ////////////////////////////////
+  // Speeding Up Percolator
+  // John T. Halloran, Hantian Zhang, Kaan Kara, Cédric Renggli, Matthew The, Ce Zhang, David M. Rocke, Lukas Käll, and William Stafford Noble
+  //   Journal of Proteome Research 2019 18 (9), 3353-3359
+  // ////////////////////////////////
+  // Note that the implementation further improves on the speedups in the paper by: 
+  //   -implementing a single threadpool using OMP for SVM training per each cpos,cneg pair per nested CV fold
+  //   -has a much smaller memory footprint by fixing memory leaks in L2_SVM_MFN and more efficient validation of the learned SVM parameters
+  // ////
+
+  // Create SVM input data for parallelization
+   std::vector<AlgIn*> svmInputsVec;
+   std::vector< std::vector< Scores > > nestedTestScoresVec;
+   for (int set = 0; set < numFolds_; ++set) {
+     std::vector<Scores> nestedTrainScores(nestedXvalBins_, usePi0_), nestedTestScores(nestedXvalBins_, usePi0_);
+     if (nestedXvalBins_ > 1) {
+       FeatureMemoryPool featurePool;
+#pragma omp ordered
+       {
+	 trainScores_[set].createXvalSetsBySpectrum(nestedTrainScores, nestedTestScores, nestedXvalBins_, featurePool);
+       }
```

Comment:
This part is currently not in an omp environment, so no need for `omp ordered`.

Reply:
Sounds good, this was from the original codebase, I removed it in the most recent commit.